### PR TITLE
Kill Unicorns

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
- Enables `unicorn-worker-killer` on all staging and production servers.
- Adds a missing lograte entry for exponential unicorn log
- Improves how ENV vars passed to unicorn service

### CODE REVIEW

File: inventory/group_vars/all.yml [hunk 1]
```diff
@@ -168,6 +180,8 @@ unicorn_pid: "{{ pid_path }}/unicorn.pid"
 unicorn_sock: "{{ sock_path }}/unicorn.{{ app }}.sock"
 unicorn_workers: 2
 unicorn_timeout: 30
+unicorn_env_vars: |
+  KILL_UNICORNS=true
```
Comment (Matt-Yorkley) [inline inventory/group_vars/all.yml:184]: Enable `unicorn-worker-killer` with the default settings on all servers (only applies to prod and staging)

File: inventory/host_vars/openfoodnetwork.org.au/config.yml [hunk 2]
```diff
@@ -33,3 +33,8 @@ custom_application_env_vars: |
   OFN_FEATURE_CONNECT_AND_LEARN: "true"
 
 enable_rails_apm: "true" # has to be explicitly defined as a string due to some datadog bug
+
+unicorn_env_vars: |
+  KILL_UNICORNS=true
+  UWK_MEM_MIN=1900
+  UWK_MEM_MAX=2100
```
Comment (Matt-Yorkley) [inline inventory/host_vars/openfoodnetwork.org.au/config.yml:40]: Apply looser settings to Aus production for now, as requested by Maikel here: https://github.com/openfoodfoundation/openfoodnetwork/pull/5162#pullrequestreview-387969686

File: roles/webserver/templates/unicorn.service.j2 [hunk 3]
```diff
@@ -7,7 +7,7 @@ Description=Open Food Network unicorn server
 Type=forking
 User=openfoodnetwork
 WorkingDirectory={{ current_path }}
-Environment=RAILS_ENV={{ rails_env }}
+EnvironmentFile={{ shared_path }}/unicorn_environment
```
Comment (sauloperez) [inline roles/webserver/templates/unicorn.service.j2:10]: :ok_hand:

Comment (Matt-Yorkley) [reply roles/webserver/templates/unicorn.service.j2:10]: As a side note here, we define some ENV vars for the openfoodnetwork user in `/etc/default/openfoodnetwork`, but as far as I can tell they don't get refreshed when they're re-provisioned. We should re-assess that soon and stop doing it / improve it. It's a separate issue though.

Comment (luisramos0) [reply roles/webserver/templates/unicorn.service.j2:10]: I'd add a comment to this line briefly explaining why RAILS_ENV doesnt work.

**Comment by Matt-Yorkley** [comment]
Tested on `fr-staging` with this and the associated PR. It's working as intended :+1:

**Pull request merged**